src/usermod.c

```diff
@@ -1957,7 +1957,7 @@ static void update_lastlog (void)
 			/* Reset the new uid's lastlog entry */
 			memzero (&ll, sizeof (ll));
 			if (   (lseek (fd, off_newuid, SEEK_SET) != off_newuid)
-			    || (write (fd, &ll, sizeof ll) != (ssize_t) sizeof ll)
+			    || (write_full (fd, &ll, sizeof ll) != (ssize_t) sizeof ll)
```

Comment:
Another interesting thing of this API, compared to write(2), is that it either writes it all, or fails.  So, having the written size in the return value is rather useless.  We could instead just return either 0 or -1.